src/domain/accounts/entities/create-account.dto.entity.ts

```diff
@@ -1,12 +1,26 @@
-import type { CreateAccountDtoSchema } from '@/domain/accounts/entities/schemas/create-account.dto.schema';
-import type { z } from 'zod';
+import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import { z } from 'zod';
+
+export const CreateAccountDtoSchema = z.object({
+  address: AddressSchema,
+  name: z
+    .string()
+    .min(3, { message: 'Account names must be at least 3 characters long' })
+    .max(20, { message: 'Account names must be at most 20 characters long' })
+    .regex(/^[a-zA-Z0-9]+(?:[._-][a-zA-Z0-9]+)*$/, {
+      message:
+        'Account names must start with a letter or number and can contain alphanumeric characters, periods, underscores, and hyphens',
+    }),
+});
```

Comment:
As before regarding extracting. We could have a `AccountNameSchema`, etc. that is tested and extended for this and the `AccountSchema`.

Reply:
Done in [2069930](https://github.com/safe-global/safe-client-gateway/pull/2081/commits/2069930ce18518a38b763ad69b1a7585e6f2016e)